contracts/adapters/voting/OffchainVoting.sol

```diff
@@ -300,6 +252,64 @@ contract OffchainVotingContract is
         vote.resultRoot = resultRoot;
         vote.reporter = memberAddr;
         vote.isChallenged = false;
+
+        emit VoteResultSubmitted(
+            proposalId,
+            result.nbNo,
+            result.nbYes,
+            resultRoot,
+            memberAddr
+        );
+    }
+
+    function requestStep(
+        DaoRegistry dao,
+        bytes32 proposalId,
+        uint256 index
+    ) external {
+        address memberAddr = dao.getAddressIfDelegated(msg.sender);
+        Voting storage vote = votes[address(dao)][proposalId];
+        require(isActiveMember(dao, memberAddr), "not active member");
+        uint256 currentFlag = retrievedStepsFlags[vote.resultRoot][index / 256];
+        require(
+            getFlag(currentFlag, index % 256) == false,
+            "step already requested"
+        );
+        require(vote.stepRequested == 0, "other step already requested");
+        vote.stepRequested = index;
+        vote.gracePeriodStartingTime = block.timestamp;
```

Comment:
I can see we verify if the step was already requested, but was wondering if we need to set the flag to true after the grace period, so we ensure that step won't be requested again.

Reply:
good catch. fixed this

Reply:
Looks great!